pkgs/development/tools/revive/default.nix

```
@@ -0,0 +1,59 @@
+{ buildGoModule, fetchFromGitHub, lib }:
+
+buildGoModule rec {
+  pname = "revive";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "mgechev";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-0kHO5/vZ+EfkeJ+BtA7N9mLwU0BFp1xfGv5ZuQByfzY=";
+    # populate values that require us to use git. By doing this in postFetch we
+    # can delete .git afterwards and maintain better reproducibility of the src.
+    leaveDotGit = true;
+    postFetch = ''
+      git rev-parse HEAD > $out/COMMIT
+      # 0000-00-00T00:00:00Z
+      date -u -d "@$(git log -1 --pretty=%ct)" "+%Y-%m-%dT%H:%M:%SZ" > $out/SOURCE_DATE_EPOCH
```

```suggestion
      # '0000-00-00 00:00 UTC'
      date -u -d "@$(git log -1 --pretty=%ct)" "'+%Y-%m-%d %H:%M UTC'" > $out/SOURCE_DATE_EPOCH
```

SHA will need updating